pyxem/utils/correlation_utils.py

```diff
@@ -151,5 +151,73 @@ def _pearson_correlation(z, mask=None):
     return p_correlation
 
 
+def _wrap_set_float(target, bottom, top, value):
+    """This function sets values in a list assuming that
+    the list is circular and allows for float bottom and float top
+    which are equal to the residual times that value.
+    """
+    ceiling_bottom = int(np.ceil(bottom))
+    residual_bottom = ceiling_bottom-bottom
+    floor_top = int(np.floor(top))
+    residual_top = top-floor_top
+    if floor_top > len(target) - 1:
+        target[ceiling_bottom:] = value
+        new_floor_top = floor_top % len(target)
+        target[new_floor_top] = value
+        target[new_floor_top+1] = value*residual_top
+    elif ceiling_bottom < 0:
+        target[:floor_top] = value
+        target[ceiling_bottom:] = value
+        target[ceiling_bottom-1] = value*residual_bottom
+    else:
+        target[ceiling_bottom:floor_top+1] = value
+        target[ceiling_bottom-1] = value*residual_bottom
+        if floor_top + 1 > len(target) - 1:
+            target[0] = value*residual_top
+        else:
+            target[floor_top+1] = value*residual_top
+    return target
+
+
+def _get_interpolation_matrix(angles, angular_range, num_points, method="average"):
+    """Gets an interpolation matrix which when a matrix multiplied gets the symmetry elements
+    for a correlation.
+    """
+    if method is "average":
+        angular_ranges = [(angle - angular_range / (2*np.pi),
+                           angle + angular_range / (2*np.pi)) for angle in angles]
+        angular_ranges = np.multiply(angular_ranges, num_points)
+        angular_ranges = np.subtract(angular_ranges, 0.5)
```

Comment:
I'm not sure why angular_ranges are shifted by half a pixel here.

Reply:
The shift by 1/2 is to deal with the fact that the middle of the pixel is actually the center. So for 90 pixels in the azimuthal direction unity is actually centered at 0.5.   For the angular range this is important because otherwise things are not properly centered and you end up sampling a little bit of a lopsided angular range.

Reply:
So I have this tested now. It seems like I was accounting for this shift in two different places.  The `wrap_set_float` function mostly handles this and we don't need to add or subtract (as you first suggested!).  With the new added tests this should be checked and better tested.